src/libsyntax/parse/parser.rs

```diff
@@ -4700,7 +4700,7 @@ impl<'a> Parser<'a> {
                 if macro_legacy_warnings && self.token != token::Semi {
                     self.warn_missing_semicolon();
                 } else {
-                    self.expect_one_of(&[token::Semi], &[])?;
+                    self.expect_one_of(&[], &[token::Semi])?;
```

Comment:
OK, this was non-obvious, but it looks like there are no cases in which this would change observable behavior.
"No `;`", "1 `;`", "2 `;`" and ">2 `;`" should all be treated same due to contexts in which `parse_full_stmt` is used.